syntax = "proto3";

package books;
option go_package = "github.com/MaximeBoulle/OOS-Projet/services/gRPC/proto;proto";

import "google/api/annotations.proto";

service BookService {
    rpc GetBook (BookRequest) returns (BookResponse) {
        option (google.api.http) = {
            get: "/v1/books/{id}"
        };
    }
    rpc GetAllBooks (Empty) returns (BooksResponse) {
        option (google.api.http) = {
            get: "/v1/books"
        };
    }
}

message BookRequest {
    int32 id = 1;
}

message BookResponse {
    int32 id = 1;
    string title = 2;
    string author = 3;
    string isbn = 4;
    float average_rating = 5;
}

message BooksResponse {
    repeated BookResponse books = 1;
}

message Empty {}